### ISSUE TITLE
Unable to use `Date` type in interfaces that will be validated

### ISSUE THREAD

**Issue by sarah-buhler** [woutervh-/typescript-is#31]
While using this library I have run into an issue where any interface I would like to be validated cannot hold any properties that hold a `Date` type. I have both `"ignoreClasses": true` and `"ignoreMethods": true` in my `tsconfig.json`. This problem occurs at run time, as validations with `Date` types fail every time, though it compiles just fine.

Glancing through the generated code, it seems a large list of checks is created for each `Date` typed property, that checks that it has a number of properties, including properties that aren't always supported. I found the first place that fails is a check that a `Date` has the `getVarDate()` method, which is only supported in Internet Explorer and obsolete according to MDN. I have not checked the rest. I am using this on server application for cursory incoming type checking.

Right now my workaround is to have a separate interface without any `Date` properties simply for use with validation and then checking for those properties manually. This would not work with the setting `"disallowSuperfluousObjectProperties": true` in the `tsconfig.json`.

**Comment by woutervh-** [comment]
Hi @sarah-buhler 

Thanks for the detailed report. Classes (especially Date and RegExp) have been a thorn in my side for this project. The typical use case for `typescript-is` is that can be used as part of some deserialization process. Often people are using it on object that also have classes for example in your situation. Classes can't (easily) be checked as of right now (see discussion #20).

With classes such as `Date` and `RegExp` there is the situation that TypeScript defines them using interfaces, not classes, which makes the situation even more difficult to deal with.

Anyway long story short: I need to find some better way of detecting classes even when they're defined using interfaces so that they can be skipped with `ignoreClasses`. I'll try to come up with something and post updates here.

**Comment by woutervh-** [comment]
hi @sarah-buhler 

I have made an attempt at detecting classes such as `Date` and `RegExp` better.
Please try it:
https://github.com/woutervh-/typescript-is/releases/tag/v0.12.3
